lib/Backend/arm64/LowerMD.cpp

```diff
@@ -2089,9 +2090,30 @@ LowererMD::ChangeToAssign(IR::Instr * instr, IRType type)
             instr->ReplaceSrc1(instr->GetSrc1()->UseWithNewType(TyFloat64, instr->m_func));
         }
     }
+    else if ((IRType_IsSignedInt(destType) || IRType_IsUnsignedInt(destType)) && TySize[destType] > TySize[srcType])
```

Comment:
>TySize[destType] > TySize[srcType] [](start = 81, length = 34)

nit: Move this check first as most of time this check will fail and we won't have to do IRType check

Reply:
Done.